### PR TITLE
 Minor: Port more aggregate tests to sqllogictests

### PR DESCRIPTION
**Pull request by alamb**
# Which issue does this PR close?

Part of https://github.com/apache/arrow-datafusion/issues/4495

Follow on to https://github.com/apache/arrow-datafusion/pull/5434

# Rationale for this change

I am trying to keep the testability of DataFusion reasonable. Sqllogictests are easier to add / update so we are trying to move stuff over there.

# What changes are included in this PR?

Port more aggregate tests to aggregate.slt

# Are these changes tested?
Yes all tests

# Are there any user-facing changes?
No

### CODE REVIEW

File: datafusion/core/tests/sqllogictests/test_files/aggregate.slt [hunk 1]
```diff
@@ -1284,3 +1284,148 @@ NULL 2
...
+ ('2018-11-13T17:11:10.011375885995', 'Row 0'),
+ ('2011-12-13T11:13:10.12345', 'Row 1'),
+ (null, 'Row 2'),
+ ('2021-1-1T05:11:10.432', 'Row 3');
+
+
+statement ok
+create table t as
+select
+  arrow_cast(column1, 'Timestamp(Nanosecond, None)') as nanos,
```
Comment (alamb) [inline datafusion/core/tests/sqllogictests/test_files/aggregate.slt:1308]: I added `arrow_cast` in  #5166 to help port this kind of test (to control the arrow types explicitly)

File: datafusion/core/tests/sqllogictests/test_files/aggregate.slt [hunk 2]
```diff
@@ -1284,3 +1284,148 @@ NULL 2
...
+query PPPPT
+select * from t;
+----
+2018-11-13T17:11:10.011375885 2018-11-13T17:11:10.011375 2018-11-13T17:11:10.011 2018-11-13T17:11:10 Row 0
+2011-12-13T11:13:10.123450 2011-12-13T11:13:10.123450 2011-12-13T11:13:10.123 2011-12-13T11:13:10 Row 1
+NULL NULL NULL NULL Row 2
+2021-01-01T05:11:10.432 2021-01-01T05:11:10.432 2021-01-01T05:11:10.432 2021-01-01T05:11:10 Row 3
+
+
+# aggregate_timestamps_sum() -> Result<()> {
```
Comment (korowa) [inline datafusion/core/tests/sqllogictests/test_files/aggregate.slt:1325]: I guess the part with function signature is excessive here

File: datafusion/core/tests/sqllogictests/test_files/aggregate.slt [hunk 3]
```diff
@@ -1284,3 +1284,148 @@ NULL 2
...
+18:06:30.243620451 18:06:30.243620 18:06:30.243 18:06:30
+
+# aggregate_times_max
+query DDDD
+SELECT max(nanos), max(micros), max(millis), max(secs) FROM t
+----
+21:06:28.247821084 21:06:28.247821 21:06:28.247 21:06:28
+
+
+# aggregate_times_avg
```
Comment (korowa) [inline datafusion/core/tests/sqllogictests/test_files/aggregate.slt:1421]: 👍

File: datafusion/core/tests/sqllogictests/test_files/aggregate.slt [hunk 4]
```diff
@@ -1284,3 +1284,148 @@ NULL 2
...
+2011-12-13T11:13:10.123450 2011-12-13T11:13:10.123450 2011-12-13T11:13:10.123 2011-12-13T11:13:10 Row 1
+NULL NULL NULL NULL Row 2
+2021-01-01T05:11:10.432 2021-01-01T05:11:10.432 2021-01-01T05:11:10.432 2021-01-01T05:11:10 Row 3
+
+
+# aggregate_timestamps_sum() -> Result<()> {
+statement error Error during planning: The function Sum does not support inputs of type Timestamp\(Nanosecond, None\)
+SELECT sum(nanos), sum(micros), sum(millis), sum(secs) FROM t;
+
+query IIII
```
Comment (korowa) [inline datafusion/core/tests/sqllogictests/test_files/aggregate.slt:1329]: Maybe it's wort adding test name for this case

**Review by korowa — APPROVED** [review]

File: datafusion/core/tests/sqllogictests/test_files/aggregate.slt (hunk 2 above)
Comment (alamb) [reply datafusion/core/tests/sqllogictests/test_files/aggregate.slt]: Agreed -- removed in [9f59a70](https://github.com/apache/arrow-datafusion/pull/5574/commits/9f59a7002e681de2aa6b7e81f0e6a546f6821546)

File: datafusion/core/tests/sqllogictests/test_files/aggregate.slt (hunk 4 above)
Comment (alamb) [reply datafusion/core/tests/sqllogictests/test_files/aggregate.slt:1330]: Good idea -- added in 5f77bb002

**Review by jackwener — APPROVED** [review]

**Pull request merged**